### ISSUE TITLE
About some custom caffe layer ?

### ISSUE THREAD

**Issue by yuzhegao** [NVlabs/ssn_superpixels#22]
Hi, @varunjampani 
Thanks for sharing your perfect code, I have run this training code and get good result, But I 'm not very familiar to caffe , so could you please tell me the function of custom caffe layer 'L.RelToAbsIndex' and 'L.Smear' ?

**Comment by varunjampani** [comment]
The network produces 'relative' superpixel indices at each pixel. That is, at each pixel, it produces a value in [0, 8] denoting one of the surrounding 9 superpixels. RelToAbsIndex layer is used to convert these relative superpixel indices to absolute indices. 'Smear' layer converts superpixel features into pixel features. This layer assign given superpixel feature to each pixel within that superpixel.

**Comment by yuzhegao** [comment]
Thanks for your answer! 
Besides, I want to  add the ssn module in CNN network of my own  to get benefit from the clustered feature. However, I found that the magnitude of the loss is not match (i.e.  the compactness-loss + reconstruction loss is around 1.5 , but I just use a cross-entropy loss without 'mean', which is around 2000), could you please share some advice to solver this problem? 
Currently I just add a very large loss-weight (1000)  to make the magnitude of this two loss items match. But I don' think this is a good approach.
Thanks in advance ! :)

**Comment by varunjampani** [comment]
I did not remember the magnitude of loss values. But, scaling them to match the magnitude is a reasonable thing to try.
